### PR TITLE
[RCT-89] React Router Basics

### PR DESCRIPTION
**Pull request by arvin-viar**
# Related Tickets

https://codeandtheory.atlassian.net/browse/RCT-89

# Description

Please merge after PR #5 

This PR implements react router basics.

# Validation

https://user-images.githubusercontent.com/77320674/195344891-81b8e0f8-1098-4283-901d-cc4705b15749.mov

# Risks

### CODE REVIEW

File: src/components/RouterCourse/Courses/CourseContainer.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import Course from './Course';
+
+const CourseContainer = ({data}) => {
+  let courses = data.map((course) => {
+    return <Course title={course.title}
+                   desc={course.description}
+                   img={course.img_src}
+                   key={course.id} />
+  });
```
Comment (lycoangelo) [inline src/components/RouterCourse/Courses/CourseContainer.js]: - `let` variable values were not reassigned. We can replace it with const instead.
- we can implement implicit return here.
- also, indentation seems off to React standard.

File: src/components/RouterCourse/Courses/CourseContainer.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <div>
+      <ul>
+        {courses}
```
Comment (lycoangelo) [inline src/components/RouterCourse/Courses/CourseContainer.js]: Actually, we can insert the `data` map here instead of assigning it to a variable so we can implement an implicit return.

File: src/components/RouterCourse/Teachers/index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        <p>{teacher.bio}</p>
+      </li>
+    );
+  });
+
+  return (
+    <div className="main-content">
+      <h2>Teachers</h2>
+      <ul className="group">
+        {teachers}
```
Comment (lycoangelo) [inline src/components/RouterCourse/Teachers/index.js]: The same here. We can insert the `TeacherList` map here instead of assigning it to a variable so we can implement an implicit return.

File: src/pages/App/App.js [hunk 4]
```diff
@@ -3,28 +3,73 @@ import {
...
-        <Route
-          path="/scoreboard"
-          element={<ScoreBoardPage />}
-        />
-      </Routes>
-    </Router>
-  </div>
-);
+const App = () => {
+  return (
```
Comment (lycoangelo) [inline src/pages/App/App.js]: We can implement implicit return here.

**Review by lycoangelo — CHANGES_REQUESTED** [review]

File: src/components/RouterCourse/Courses/CourseContainer.js (hunk 2 above)
Comment (arvin-viar) [reply src/components/RouterCourse/Courses/CourseContainer.js]: updated with the following changes
- moved data mapping inside ul tag
- apply implicit return
- fixed indentation

File: src/components/RouterCourse/Teachers/index.js (hunk 3 above)
Comment (arvin-viar) [reply src/components/RouterCourse/Teachers/index.js]: updated with the following changes
- moved data mapping inside ul tag
- apply implicit return
- fixed indentation

File: src/pages/App/App.js (hunk 4 above)
Comment (arvin-viar) [reply src/pages/App/App.js]: updated, applied implicit return

File: src/components/RouterCourse/Courses/index.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    {course === "html" && (
+      <CourseContainer data={HTMLCourses} />
+    )}
+    {course === "css" && (
+      <CourseContainer data={CSSCourses} />
+    )}
+    {course === "javascript" && (
+      <CourseContainer data={JSCourses} />
+    )}
```
Comment (lycoangelo) [inline src/components/RouterCourse/Courses/index.js]: We can shorten this to something like this:

```
const dataMap = {
  html: HTMLCourses,
  css: CSSCourses,
  javascript: JSCourses,
};
```

and then just write: `<CourseContainer data={dataMap[course]} />`

File: src/pages/RouterCourse/index.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        )}
+        {page === "courses" && (
+          <Courses course={course} />
+        )}
+      </div>
+    </section>
+  </div>
+);
+
+export default RouterCoursePage;
```
Comment (lycoangelo) [inline src/pages/RouterCourse/index.js:29]: The same thing here. I think we can shorten this into something like this. Add this before the function component:
```
const componentMap = {
  home: Home,
  about: About,
  teachers: Teachers,
  courses: Courses,
};
```
and then just write: `{componentMap[page](course)}`

**Review by lycoangelo — CHANGES_REQUESTED** [review]

File: src/pages/RouterCourse/index.js (hunk 6 above)
Comment (arvin-viar) [reply src/pages/RouterCourse/index.js:29]: updated, applied the requested code change

File: src/components/RouterCourse/Courses/index.js (hunk 5 above)
Comment (arvin-viar) [reply src/components/RouterCourse/Courses/index.js]: updated, applied the requested code change

File: src/pages/RouterCourse/index.js (hunk 6 above)
Comment (lycoangelo) [reply src/pages/RouterCourse/index.js:29]: The `componentMap` can be placed outside the function component so it won't be redeclared every time the component re-renders.

Then, we can implement implicit return in this component.

File: src/components/RouterCourse/Courses/index.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+import CourseContainer from './CourseContainer';
+import { HTMLCourses, CSSCourses, JSCourses } from '../data/Courses';
+
+const Courses = ({course}) => {
+  const courseDataMap = {
+    html: HTMLCourses,
+    css: CSSCourses,
+    javascript: JSCourses,
+  };
```
Comment (lycoangelo) [inline src/components/RouterCourse/Courses/index.js]: The `courseDataMap ` can be placed outside the function component so it won't be redeclared every time the component re-renders.

Then, we can also implement implicit return in this component.

Comment (arvin-viar) [reply src/components/RouterCourse/Courses/index.js]: updated, code change are now applied

**Review by lycoangelo — APPROVED** [review]
